jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/content/PathContentSource.java

```diff
@@ -84,22 +89,30 @@ public long getLength()
 
     public int getBufferSize()
     {
-        return bufferSize;
+        return byteBufferPool.getSize();
     }
 
     public void setBufferSize(int bufferSize)
     {
-        this.bufferSize = bufferSize;
+        try (AutoLock ignored = lock.lock())
+        {
+            if (bufferSize != byteBufferPool.getSize())
+                byteBufferPool = new ByteBufferPool.Sized(byteBufferPool.getWrapped(), byteBufferPool.isDirect(), bufferSize);
+        }
     }
 
     public boolean isUseDirectByteBuffers()
     {
-        return useDirectByteBuffers;
+        return byteBufferPool.isDirect();
     }
 
     public void setUseDirectByteBuffers(boolean useDirectByteBuffers)
```

Comment:
Should I deprecate the direct and buffer size setters in favour of the constructor taking a Sized pool?

Reply:
+1